src/requests/models.py

```diff
@@ -663,10 +665,10 @@ def __init__(self):
         #: Integer Code of responded HTTP Status, e.g. 404 or 200.
         self.status_code = None
 
-        #: Case-insensitive Dictionary of Response Headers.
+        #: Case-insensitive Dictionary (email.EmailMessage) of Response Headers.
         #: For example, ``headers['content-encoding']`` will return the
         #: value of a ``'Content-Encoding'`` response header.
-        self.headers = CaseInsensitiveDict()
```

Comment:
This is specifically backwards-incompatible. `EmailMessage` and the `CaseInsensitiveDict` do not have compatible methods.

Reply:
Indeed, if somebody wanted to modify the provided object the modification methods are not compatible.

Although such use is not tested if it's possible to do somebody is bound to do it.

What is an option to provide correct headers?

Save the original response headers in another property?